### PR TITLE
Fix PlatformDispatcher.locale to return something meaningful when there are no locales.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

Returns an "undefined" locale (language code "und") from `PlatformDispatcher.locale` when no locales are defined.

## Tests

 - Added tests to web and regular versions.

## Breaking Change

- [X] No, no existing tests failed, so this is *not* a breaking change.

### CODE REVIEW

File: testing/dart/window_hooks_integration_test.dart [hunk 1]
```diff
@@ -426,4 +426,16 @@ void main() {
...
+  test('PlatformDispatcher.locale returns unknown locale when locales is set to empty list', () {
+    Locale? locale;
+    runZoned(() {
+      window.onLocaleChanged = () {
+        locale = PlatformDispatcher.instance.locale;
+      };
+    });
+
+    _updateLocales(<String>[]);
+    expectEquals(locale, const Locale.fromSubtags());
```
Comment (HansMuller) [inline testing/dart/window_hooks_integration_test.dart:439]: The API docs say that we return a Locale with language tag 'und', but https://api.flutter.dev/flutter/package-intl_locale/Locale/fromSubtags.html doesn't seem to guarantee that. Maybe we should be checking as much here?

**Review by HansMuller — COMMENTED** [review]
Is there downstream code that consumes locales which must be updated to deal with the "undefined" locale rather than null?

File: testing/dart/window_hooks_integration_test.dart (hunk 1 above)
Comment (gspencergoog) [reply testing/dart/window_hooks_integration_test.dart:439]: Or we could change the docs for `fromSubtags`...

**Comment by gspencergoog** [comment]
No, there's only one downstream usage that seemed to care that the value might be null, and this PR is to fix that usage.

File: testing/dart/window_hooks_integration_test.dart (hunk 1 above)
Comment (gspencergoog) [reply testing/dart/window_hooks_integration_test.dart:439]: I did both.

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**